spec.html

```diff
@@ -437,9 +435,9 @@ <h1>Iterator.prototype.drop ( _limit_ )</h1>
               1. Let _next_ be ? IteratorStep(_iterated_).
               1. If _next_ is *false*, return *undefined*.
             1. Repeat,
-              1. Let _next_ be ? IteratorStep(_iterated_).
-              1. If _next_ is *false*, return *undefined*.
-              1. Let _completion_ be Completion(Yield(? IteratorValue(_next_))).
+              1. Let _value_ be ? IteratorStepValue(_iterated_).
+              1. If _value_ is ~done~, return *undefined*.
+              1. Let _completion_ be Completion(Yield(_value_)).
```

Comment:
i think this has slightly different semantics - currently, if IteratorStep is abrupt, it does not close the iterator - but with these changes, it does. Is that correct?

Reply:
No, with this change, it still  does not close the iterator. If IteratorStep is abrupt, then IteratorStepValue will be abrupt, and so the `Let _value_ be ? IteratorStepValue(_iterated_)` line will early exit (because of the `?`).

Reply:
right - but if the IteratorValue part is abrupt, it *wont* close the iterator.

Reply:
That was true before and is true now.

In the original, an abrupt completion for `IteratorValue` in `? IteratorValue(_next_)` in the original will cause the `Let _completion_ be Completion(Yield(? IteratorValue(_next_)))` line to early exit before getting to the `IfAbruptCloseIterator`.

In the new version, the abrupt completion would happen in `? IteratorStepValue(_iterated_)`, with the same effect.

Reply:
ahh ok thanks, i was uncertain but that's convincing. I'll continue implementing and reviewing the rest.